src/request/transaction.ts

```diff
@@ -117,21 +117,18 @@ export default class Transaction {
   private sanitizeKeys() {
     const sanitized = Object.assign({}, this) as any;
 
-    if (
-      sanitized.creditCard != null &&
-      sanitized.creditCard.last4digits != null
-    ) {
-      sanitized.creditCard.last_4_digits = this.creditCard?.last4digits;
+    if (sanitized.creditCard?.last4digits) {
+      sanitized.creditCard.last_4_digits = sanitized.creditCard.last4digits;
       delete sanitized.creditCard.last4digits;
     }
 
-    if (sanitized.billing != null && sanitized.billing.address2 != null) {
-      sanitized.billing.address_2 = this.billing?.address2;
+    if (sanitized.billing?.address2) {
```

Comment:
Do we still want to compare against `null` so that we handle the empty string and 0 in the same way? That said, I wonder if it would be even better to check if the key exists here rather than checking the value. I believe we could end up with a sanitized object that still had `address2` with a null value. (Note that I haven't checked all the code leading to this point, so it is possible something else prevents that.)

Same for the others.

Reply:
Sorry Greg, I'm a bit confused by this. Here are a few thoughts I have.

1. TypeScript is [handling data-validation elsewhere](https://github.com/maxmind/minfraud-api-node/blob/main/src/request/location.ts#L23). `address2` can only be `undefined` or a `string`. Defining `address2` as `null` throws an error. The same applies for the other properties.

```
src/request/transaction.spec.ts:324:13 - error TS2322: Type 'null' is not assignable to type 'string | undefined'.

    324             address2: null,
                    ~~~~~~~~

      src/request/location.ts:23:3
        23   address2?: string;
             ~~~~~~~~
        The expected type comes from property 'address2' which is declared here on type 'LocationProps'
```

2. The purpose of this "sanitization" is to transform camel-cased property names containing numbers to kebab-cased property names. I'm thinking that this function should remain dumb and only have one job – mapping the value of the camel-cased key to the kebab-cased key, regardless if the value is valid or not. If there is a data validation issue, that should be solved elsewhere. 

Am I misunderstanding your ask here? Also, is the function name "sanitization" a point of confusion? Should it be named a bit more verbose to reflect what this chunk of code is doing?

Reply:
I guess it is fine. I believe it does mean that we may be sending `address2` to the web service when used from plain JS, which will result in a somewhat confusing warning. This is why I didn't see it as a validation issue, but an issue with the fixing of the camelcase. If `address_2` was sent with a `null` value, you would get a more sensible warning about the input being invalid.

Reply:
@oschwald I pushed an additional commit that changes the checks from an explicit null check to just checking for the existence of keys before converting them. I also added some tests.

If this wasn't what you were intending, just let me know.